src/disk_utils.cpp

```diff
@@ -1047,21 +1050,22 @@ int build_disk_index(const char *dataFilePath, const char *indexFilePath, const
     {
         param_list.push_back(cur_param);
     }
-    if (param_list.size() < 5 || param_list.size() > 8)
-    {
-        diskann::cout << "Correct usage of parameters is R (max degree)\n"
-                         "L (indexing list size, better if >= R)\n"
-                         "B (RAM limit of final index in GB)\n"
-                         "M (memory limit while indexing)\n"
-                         "T (number of threads for indexing)\n"
-                         "B' (PQ bytes for disk index: optional parameter for "
-                         "very large dimensional data)\n"
-                         "reorder (set true to include full precision in data file"
-                         ": optional paramter, use only when using disk PQ\n"
-                         "build_PQ_byte (number of PQ bytes for inde build; set 0 to use "
-                         "full precision vectors)"
-                      << std::endl;
-        return -1;
+    if (param_list.size() < 5 || param_list.size() > 9) {
+      diskann::cout
+          << "Correct usage of parameters is R (max degree)\n"
+             "L (indexing list size, better if >= R)\n"
+             "B (RAM limit of final index in GB)\n"
+             "M (memory limit while indexing)\n"
+             "T (number of threads for indexing)\n"
+             "B' (PQ bytes for disk index: optional parameter for "
+             "very large dimensional data)\n"
+             "reorder (set true to include full precision in data file"
+             ": optional paramter, use only when using disk PQ\n"
+             "build_PQ_byte (number of PQ bytes for inde build; set 0 to use "
+             "full precision vectors)\n"
+             "QD Quantized Dimension"
```

Comment:
need some logic and documentation around how it interacts with the memory limit, `B` -- will quantized dimension only apply if it fits inside the memory limit, or does it override it?

Reply:
sure, can do.